x-pack/plugins/observability_solution/observability_ai_assistant/server/service/client/index.ts

```diff
@@ -495,11 +505,12 @@ export class ObservabilityAIAssistantClient {
               conversation: {
                 title: generatedTitle || title || 'New conversation',
                 token_count: tokenCountResult,
+                id: conversationId || '',
```

Comment:
Setting to an empty string feels like a hack. If we want to make it optional, then let's change the type.